aws/config.go

```diff
@@ -265,6 +265,8 @@ type Config struct {
 	//    	Bucket: aws.String("bucketname"),
 	//    	Key: aws.String("//foo//bar//moo"),
 	//    })
+	// Deprecated: DisableRestProtocolURICleaning exists for historical compatibility of
```

Comment:
Since this setting is now effectively useless, let's remove everything but the deprecation notice. Let's also reword to clarify the behavior now:
`Deprecated: This setting has no longer has any effect. RESTful paths are no longer cleaned after request serialization.`